File: Project 1 Question 4/src/Animal.java
/**
 * 
 * This is the main class Animal which will extended by other classes
 *
 */

public class Animal
{
	protected String name;
	
	/**
	 * Constructor to set the name of the Animal
	 * @param name - Name of the Animal
	 */
	public Animal(String name)
	{
		this.name = name;
	}
	
	/**
	 * Method which ask animal to sleep
	 */
	public void sleep()
	{
		System.out.println(this.name+" the "+this.getClass().getName() );
		System.out.println(this.name+" slept");
		
	}
	/**
	 * Method to get the animal eat
	*/
	public void eat()
	{
		System.out.println(this.name+" the "+this.getClass().getName() );
		System.out.println(this.name+" ate meat");
	}
	/**
	 * Method to wake up the animal
	 */
	public void wakeUp()
	{
		System.out.println(this.name+" the "+this.getClass().getName() );
		System.out.println(this.name+" woke up");
	}
	
	/**
	 * Method which tell animal to exercise
	 */
	
	public void roam()
	{
		System.out.println(this.name+" the "+this.getClass().getName() );
		System.out.println(this.name+" ran around");
	}
	
	/**
	 * Method to get the animal make noise
	 */
	public void makeNoise()
	{
		System.out.println(this.name+" the "+this.getClass().getName() );
		System.out.println(this.name+" making noises");
	}
	
}

File: Project 2/Animal.cpp
#include "Animal.h"

#include<iostream>
using namespace std;

Animal::Animal(string name,string type)
{
  name_=name;
  type_=type;
}
void Animal::Sleep()
{
  cout<<name_<<" is a "<<type_<<endl;
  cout<<name_<<" slept"<<endl;
}

void Animal::WakeUp()
{
  cout<<name_<<" is a "<<type_<<endl;
  cout<<name_<<" woke up"<<endl;
}

void Animal::StartEat()
{
eat_->Eat(name_,type_);
}

void Animal::SetEat(AbstractEat *e)
{
  eat_=e;
}